core/src/main/java/org/hibernate/ogm/cfg/impl/HibernateSearchIntegration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.cfg.impl;
+
+import java.util.Map;
+import java.util.Properties;
+
+import org.hibernate.cfg.Configuration;
+
+/**
+ * Hibernate Search is an optional dependency, so while we need refer to its constants
+ * to make sure its configured correctly for OGM by default, we don't want to create a
+ * hard dependency. This class should be used as a helper to centralise that.
+ *
+ * @author Sanne Grinovero
+ */
+public final class HibernateSearchIntegration {
+
+	private static final boolean searchIsAvailable = isSearchAvailable();
+	private static String MASSINDEXER_PROPERTY_KEY = null;
```

Comment:
Yes, I like the centralization part.

Only I'm wondering why you are not working with the actual constants (`o.h.s.c.Environment.DATABASE_RETRIEVAL_METHOD` etc.) within this class directly but rather go through this indirection. All the usages are guarded by `if( isSearchAvailable )` so I don't see what you gain by that?

>  Hibernate Search is not a mandatory dependency: for example Neo4J needs to exclude it.

Of course. But in which way do these constants help? Also I don't think you need to avoid importing these classes. You'd only get a CNFE upon actually using them (which you nicely guarded already).